### PR TITLE
test(store): add unit tests for PG entity conversion functions

### PR DESCRIPTION
**Pull request by otavio**
## Summary

- Add comprehensive table-driven unit tests for all `ToModel` and `FromModel` conversion functions in `api/store/pg/entity/`
- 11 new test files covering tag, private-key, api-key, membership, membership-invitation, user, namespace, device, public-key, session, and system entities
- Tests cover field mapping, default value handling, nil pointer safety, and edge cases (including the nil Namespace regression in DeviceToModel from #5859)

Companion PR: shellhub-io/cloud — `test/pg-entity-conversion-tests`

## Test plan

- [x] `./bin/docker-compose exec api go test -v ./store/pg/entity/...` — 68 test cases, all passing

### CODE REVIEW

File: api/store/pg/entity/namespace_test.go [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+			assert.Equal(t, tt.expected.DevicesAcceptedCount, result.DevicesAcceptedCount)
+			assert.Equal(t, tt.expected.DevicesPendingCount, result.DevicesPendingCount)
+			assert.Equal(t, tt.expected.DevicesRejectedCount, result.DevicesRejectedCount)
+			assert.Equal(t, tt.expected.DevicesRemovedCount, result.DevicesRemovedCount)
+			assert.Equal(t, len(tt.expected.Memberships), len(result.Memberships))
+			for i, m := range tt.expected.Memberships {
+				assert.Equal(t, m.UserID, result.Memberships[i].UserID)
+				assert.Equal(t, m.NamespaceID, result.Memberships[i].NamespaceID)
+				assert.Equal(t, m.Role, result.Memberships[i].Role)
+			}
```
[the remark this answers is not in the document]
Comment (otavio) [reply api/store/pg/entity/namespace_test.go]: Already addressed in a previous commit.

File: api/store/pg/entity/namespace_test.go [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+		})
+	}
```
[the remark this answers is not in the document]
Comment (otavio) [reply api/store/pg/entity/namespace_test.go:163]: Already addressed in a previous commit.

File: api/store/pg/entity/namespace_test.go [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+			assert.Equal(t, tt.expected.Type, result.Type)
+			assert.Equal(t, tt.expected.MaxDevices, result.MaxDevices)
+			assert.NotNil(t, result.Settings, "Settings should never be nil")
+			assert.Equal(t, tt.expected.Settings.SessionRecord, result.Settings.SessionRecord)
+			assert.Equal(t, tt.expected.Settings.ConnectionAnnouncement, result.Settings.ConnectionAnnouncement)
+			assert.Equal(t, tt.expected.DevicesAcceptedCount, result.DevicesAcceptedCount)
+			assert.Equal(t, tt.expected.DevicesPendingCount, result.DevicesPendingCount)
+			assert.Equal(t, tt.expected.DevicesRejectedCount, result.DevicesRejectedCount)
+			assert.Equal(t, tt.expected.DevicesRemovedCount, result.DevicesRemovedCount)
+			assert.Equal(t, len(tt.expected.Members), len(result.Members))
```
[the remark this answers is not in the document]
Comment (otavio) [reply api/store/pg/entity/namespace_test.go]: Already addressed in a previous commit.

File: api/store/pg/entity/namespace_test.go [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+		})
+	}
```
[the remark this answers is not in the document]
Comment (otavio) [reply api/store/pg/entity/namespace_test.go:266]: Already addressed in a previous commit.

File: api/store/pg/entity/namespace_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (otavio) [reply api/store/pg/entity/namespace_test.go]: Already addressed in a previous commit.

File: api/store/pg/entity/membership-invitation_test.go [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+		t.Run(tt.name, func(t *testing.T) {
+			result := MembershipInvitationFromModel(tt.model)
+			assert.Equal(t, tt.expected.ID, result.ID)
+			assert.Equal(t, tt.expected.TenantID, result.TenantID)
+			assert.Equal(t, tt.expected.UserID, result.UserID)
+			assert.Equal(t, tt.expected.InvitedBy, result.InvitedBy)
+			assert.Equal(t, tt.expected.Role, result.Role)
+			assert.Equal(t, tt.expected.Status, result.Status)
+			assert.Equal(t, tt.expected.ExpiresAt, result.ExpiresAt)
+			assert.Equal(t, tt.expected.Invitations, result.Invitations)
```
[the remark this answers is not in the document]
Comment (otavio) [reply api/store/pg/entity/membership-invitation_test.go:115]: Already addressed in a previous commit.

File: api/store/pg/entity/membership-invitation_test.go [hunk 6]
```diff
@@ -0,0 +1,244 @@
...
+		})
```
[the remark this answers is not in the document]
Comment (otavio) [reply api/store/pg/entity/membership-invitation_test.go:119]: Already addressed in a previous commit.

File: api/store/pg/entity/membership-invitation_test.go [hunk 7]
```diff
@@ -0,0 +1,244 @@
...
+	}
```
[the remark this answers is not in the document]
Comment (otavio) [reply api/store/pg/entity/membership-invitation_test.go:120]: Already addressed in a previous commit.

File: api/store/pg/entity/membership-invitation_test.go [hunk 8]
```diff
@@ -0,0 +1,244 @@
...
+			assert.Equal(t, tt.expected.ID, result.ID)
+			assert.Equal(t, tt.expected.TenantID, result.TenantID)
+			assert.Equal(t, tt.expected.UserID, result.UserID)
+			assert.Equal(t, tt.expected.InvitedBy, result.InvitedBy)
+			assert.Equal(t, tt.expected.Role, result.Role)
+			assert.Equal(t, tt.expected.Status, result.Status)
+			assert.Equal(t, tt.expected.ExpiresAt, result.ExpiresAt)
+			assert.Equal(t, tt.expected.Invitations, result.Invitations)
+			assert.Equal(t, tt.expected.NamespaceName, result.NamespaceName)
+			assert.Equal(t, tt.expected.UserEmail, result.UserEmail)
```
[the remark this answers is not in the document]
Comment (otavio) [reply api/store/pg/entity/membership-invitation_test.go:244]: Already addressed in a previous commit.

File: api/store/pg/entity/membership-invitation_test.go [hunk 9]
```diff
@@ -0,0 +1,244 @@
...
+		})
+	}
```
[the remark this answers is not in the document]
Comment (otavio) [reply api/store/pg/entity/membership-invitation_test.go:249]: Already addressed in a previous commit.

File: api/store/pg/entity/user_test.go [hunk 10]
```diff
@@ -0,0 +1,249 @@
...
+			assert.Equal(t, tt.expected.Preferences.PreferredNamespace, result.Preferences.PreferredNamespace)
+			assert.Equal(t, tt.expected.Preferences.AuthMethods, result.Preferences.AuthMethods)
+			assert.Equal(t, tt.expected.Preferences.SecurityEmail, result.Preferences.SecurityEmail)
+			assert.Equal(t, tt.expected.Preferences.MaxNamespaces, result.Preferences.MaxNamespaces)
+			assert.Equal(t, tt.expected.Preferences.EmailMarketing, result.Preferences.EmailMarketing)
+			assert.Equal(t, tt.expected.MFA.Enabled, result.MFA.Enabled)
+			assert.Equal(t, tt.expected.MFA.Secret, result.MFA.Secret)
+			assert.Equal(t, tt.expected.MFA.RecoveryCodes, result.MFA.RecoveryCodes)
+			assert.True(t, result.UpdatedAt.IsZero(), "UpdatedAt should be zero")
+		})
```
[the remark this answers is not in the document]
Comment (otavio) [reply api/store/pg/entity/user_test.go:133]: Already addressed in a previous commit.

File: api/store/pg/entity/user_test.go [hunk 11]
```diff
@@ -0,0 +1,249 @@
...
+			assert.Equal(t, tt.expected.RecoveryEmail, result.RecoveryEmail)
+			// Password
+			assert.Equal(t, tt.expected.Password.Hash, result.Password.Hash)
+			// MFA
+			assert.Equal(t, tt.expected.MFA, result.MFA)
+			// Preferences
+			assert.Equal(t, tt.expected.Preferences, result.Preferences)
+		})
+	}
+}
```
[the remark this answers is not in the document]
Comment (otavio) [reply api/store/pg/entity/user_test.go:252]: Already addressed in a previous commit.

File: api/store/pg/entity/session_test.go [hunk 12]
```diff
@@ -0,0 +1,412 @@
...
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			result := SessionFromModel(tt.model)
+			tt.check(t, result)
+		})
+	}
+}
```
[the remark this answers is not in the document]
Comment (otavio) [reply api/store/pg/entity/session_test.go:85]: Already addressed in a previous commit.

File: api/store/pg/entity/session_test.go [hunk 13]
```diff
@@ -0,0 +1,412 @@
...
+				Session:   "session-1",
+				Type:      models.SessionEventTypePtyOutput,
+				Timestamp: now,
+				Seat:      0,
+				Data:      map[string]interface{}{"output": "hello"},
+			},
+			check: func(t *testing.T, result *SessionEvent) {
+				assert.Equal(t, "session-1", result.SessionID)
+				assert.Equal(t, "pty-output", result.Type)
+				assert.Equal(t, now, result.CreatedAt)
```
[the remark this answers is not in the document]
Comment (otavio) [reply api/store/pg/entity/session_test.go:304]: Already addressed in a previous commit.

File: api/store/pg/entity/session_test.go [hunk 14]
```diff
@@ -0,0 +1,412 @@
...
+				Session:   "session-2",
+				Type:      models.SessionEventTypeShell,
+				Timestamp: now,
+				Seat:      1,
+				Data:      nil,
+			},
+			check: func(t *testing.T, result *SessionEvent) {
+				assert.Equal(t, "session-2", result.SessionID)
+				assert.Equal(t, "shell", result.Type)
+				assert.Equal(t, "", result.Data)
```
[the remark this answers is not in the document]
Comment (otavio) [reply api/store/pg/entity/session_test.go:321]: Already addressed in a previous commit.

File: api/store/pg/entity/session_test.go [hunk 15]
```diff
@@ -0,0 +1,412 @@
...
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			result := SessionEventToModel(tt.entity)
+			tt.check(t, result)
+		})
+	}
```
[the remark this answers is not in the document]
Comment (otavio) [reply api/store/pg/entity/session_test.go:393]: Already addressed in a previous commit.

File: api/store/pg/entity/session_test.go [hunk 16]
```diff
@@ -0,0 +1,412 @@
...
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			result := parseEventSeats(tt.input)
+			assert.Equal(t, tt.expected, result)
+		})
+	}
+}
```
[the remark this answers is not in the document]
Comment (otavio) [reply api/store/pg/entity/session_test.go:471]: Already addressed in a previous commit.

File: api/store/pg/entity/device_test.go [hunk 17]
```diff
@@ -0,0 +1,253 @@
...
+				assert.Equal(t, disconnectedAt, result.DisconnectedAt)
+				assert.Equal(t, "00:11:22:33:44:55", result.MAC)
+				assert.InDelta(t, 1.23, result.Longitude, 0.001)
+				assert.InDelta(t, 4.56, result.Latitude, 0.001)
+				assert.Equal(t, "device-info-id", result.Identifier)
+				assert.Equal(t, "My Device", result.PrettyName)
+				assert.Equal(t, "1.0.0", result.Version)
+				assert.Equal(t, "amd64", result.Arch)
+				assert.Equal(t, "linux", result.Platform)
+				assert.True(t, result.UpdatedAt.IsZero())
```
[the remark this answers is not in the document]
Comment (otavio) [reply api/store/pg/entity/device_test.go:59]: Already addressed in a previous commit.

File: api/store/pg/entity/device_test.go [hunk 18]
```diff
@@ -0,0 +1,253 @@
...
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			result := DeviceToModel(tt.entity)
+			tt.check(t, result)
+		})
+	}
+}
```
[the remark this answers is not in the document]
Comment (otavio) [reply api/store/pg/entity/device_test.go:278]: Already addressed in a previous commit.

File: api/store/pg/entity/api-key_test.go [hunk 19]
```diff
@@ -0,0 +1,145 @@
...
+			assert.Equal(t, tt.expected.KeyDigest, result.KeyDigest)
+			assert.Equal(t, tt.expected.Name, result.Name)
+			assert.Equal(t, tt.expected.NamespaceID, result.NamespaceID)
+			assert.Equal(t, tt.expected.Role, result.Role)
+			assert.Equal(t, tt.expected.UserID, result.UserID)
+			assert.Equal(t, tt.expected.CreatedAt, result.CreatedAt)
+			assert.Equal(t, tt.expected.UpdatedAt, result.UpdatedAt)
+			assert.Equal(t, tt.expected.ExpiresIn, result.ExpiresIn)
+		})
+	}
```
[the remark this answers is not in the document]
Comment (otavio) [reply api/store/pg/entity/api-key_test.go:80]: Already addressed in a previous commit.

File: api/store/pg/entity/public-key_test.go [hunk 20]
```diff
@@ -0,0 +1,169 @@
...
+			check: func(t *testing.T, result *PublicKey) {
+				assert.Equal(t, "tenant-id-1", result.NamespaceID)
+				assert.Equal(t, "SHA256:abc123", result.Fingerprint)
+				assert.Equal(t, []byte("ssh-rsa AAAA..."), result.Data)
+				assert.Equal(t, "my-key", result.Name)
+				assert.Equal(t, "*.example.com", result.FilterHostname)
+				assert.Len(t, result.Tags, 2)
+				assert.Equal(t, "tag-1", result.Tags[0].ID)
+				assert.Equal(t, "prod", result.Tags[0].Name)
+				assert.True(t, result.UpdatedAt.IsZero())
```
[the remark this answers is not in the document]
Comment (otavio) [reply api/store/pg/entity/public-key_test.go:50]: Already addressed in a previous commit.

File: api/store/pg/entity/public-key_test.go [hunk 21]
```diff
@@ -0,0 +1,169 @@
...
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			result := PublicKeyFromModel(tt.model)
+			tt.check(t, result)
+		})
+	}
+}
+
+func TestPublicKeyToModel(t *testing.T) {
+	now := time.Now()
```
[the remark this answers is not in the document]
Comment (otavio) [reply api/store/pg/entity/public-key_test.go:113]: Already addressed in a previous commit.

File: api/store/pg/entity/public-key_test.go [hunk 22]
```diff
@@ -0,0 +1,169 @@
...
+				assert.Equal(t, "SHA256:abc123", result.Fingerprint)
+				assert.Equal(t, []byte("ssh-rsa AAAA..."), result.Data)
+				assert.Equal(t, "my-key", result.PublicKeyFields.Name)
+				assert.Equal(t, "", result.PublicKeyFields.Username)
+				assert.Equal(t, "*.example.com", result.Filter.Hostname)
+				assert.Len(t, result.Filter.Tags, 2)
+				assert.Equal(t, "tag-1", result.Filter.Tags[0].ID)
+				assert.Len(t, result.Filter.TagIDs, 2)
+				assert.Equal(t, "tag-1", result.Filter.TagIDs[0])
+			},
```
[the remark this answers is not in the document]
Comment (otavio) [reply api/store/pg/entity/public-key_test.go:146]: Already addressed in a previous commit.

File: api/store/pg/entity/system_test.go [hunk 23]
```diff
@@ -0,0 +1,232 @@
...
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			result := SystemToModel(tt.entity)
+			tt.check(t, result)
+		})
+	}
+}
```
[the remark this answers is not in the document]
Comment (otavio) [reply api/store/pg/entity/system_test.go:278]: Already addressed in a previous commit.

File: api/store/pg/entity/session_test.go [hunk 24]
```diff
@@ -0,0 +1,412 @@
...
+				assert.Equal(t, now, result.SeenAt)
+				assert.False(t, result.Closed)
+				assert.True(t, result.Authenticated)
+				assert.True(t, result.Recorded)
+				assert.Equal(t, "shell", result.Type)
+				assert.Equal(t, "xterm-256color", result.Term)
+				assert.InDelta(t, 1.23, result.Longitude, 0.001)
+				assert.InDelta(t, 4.56, result.Latitude, 0.001)
+				// CreatedAt and UpdatedAt use time.Now() internally
+				assert.False(t, result.CreatedAt.IsZero())
```
[the remark this answers is not in the document]
Comment (otavio) [reply api/store/pg/entity/session_test.go]: Fixed: `SessionFromModel` now calls `clock.Now()` once and reuses the value. Test uses clock mock with `t.Cleanup` to restore `DefaultBackend`.

File: api/store/pg/entity/session_test.go [hunk 25]
```diff
@@ -0,0 +1,412 @@
...
+
+	model := &models.ActiveSession{
+		UID:      "active-session-1",
+		LastSeen: now,
+	}
+
+	result := ActiveSessionFromModel(model)
+	assert.Equal(t, "active-session-1", result.SessionID)
+	assert.Equal(t, now, result.SeenAt)
+	assert.False(t, result.CreatedAt.IsZero())
```
[the remark this answers is not in the document]
Comment (otavio) [reply api/store/pg/entity/session_test.go]: Fixed: Added `t.Cleanup` to restore `clock.DefaultBackend`. Also added a zero-value edge case test.

File: api/store/pg/entity/session_test.go [hunk 26]
```diff
@@ -0,0 +1,412 @@
...
+				assert.False(t, result.Closed)
+				assert.True(t, result.Authenticated)
+				assert.True(t, result.Recorded)
+				assert.Equal(t, "shell", result.Type)
+				assert.Equal(t, "xterm-256color", result.Term)
+				assert.InDelta(t, 1.23, result.Longitude, 0.001)
+				assert.InDelta(t, 4.56, result.Latitude, 0.001)
+				// CreatedAt and UpdatedAt use time.Now() internally
+				assert.False(t, result.CreatedAt.IsZero())
+				assert.False(t, result.UpdatedAt.IsZero())
```
[the remark this answers is not in the document]
Comment (otavio) [reply api/store/pg/entity/session_test.go]: Fixed: Clock mock ensures exact value comparison via `assert.Equal`. Also added `t.Cleanup` for proper test isolation.

File: api/store/pg/entity/session_test.go [hunk 27]
```diff
@@ -0,0 +1,412 @@
...
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			result := SessionToModel(tt.entity)
+			tt.check(t, result)
+		})
+	}
```
[the remark this answers is not in the document]
Comment (otavio) [reply api/store/pg/entity/session_test.go:173]: Fixed: Added missing `Position.Latitude` assertion.

File: api/store/pg/entity/device_test.go [hunk 28]
```diff
@@ -0,0 +1,253 @@
...
+				assert.InDelta(t, 1.23, result.Position.Longitude, 0.001)
+				assert.NotNil(t, result.Info)
+				assert.Equal(t, "info-id", result.Info.ID)
+				assert.NotNil(t, result.Identity)
+				assert.Equal(t, "00:11:22:33:44:55", result.Identity.MAC)
+				assert.Len(t, result.Tags, 1)
+				assert.Equal(t, "tag-1", result.Tags[0].ID)
+				assert.Len(t, result.TagIDs, 1)
+				assert.Equal(t, "tag-1", result.TagIDs[0])
+			},
```
[the remark this answers is not in the document]
Comment (otavio) [reply api/store/pg/entity/device_test.go:210]: Fixed: Added missing `Acceptable` and `Position.Latitude` assertions.

File: api/store/pg/entity/session_test.go [hunk 29]
```diff
@@ -0,0 +1,412 @@
...
+	model := &models.ActiveSession{
+		UID:      "active-session-1",
+		LastSeen: now,
+	}
+
+	result := ActiveSessionFromModel(model)
+	assert.Equal(t, "active-session-1", result.SessionID)
+	assert.Equal(t, now, result.SeenAt)
+	assert.False(t, result.CreatedAt.IsZero())
+}
```
[the remark this answers is not in the document]
Comment (otavio) [reply api/store/pg/entity/session_test.go:221]: The function IS table-driven. Added a zero-value edge case test to improve coverage.

File: api/store/pg/entity/device_test.go [hunk 30]
```diff
@@ -0,0 +1,271 @@
...
```
[the remark this answers is not in the document]
Comment (otavio) [reply api/store/pg/entity/device_test.go:210]: Fixed: Added `assert.False(t, result.Acceptable)` assertion.

File: api/store/pg/entity/session_test.go [hunk 31]
```diff
@@ -0,0 +1,455 @@
...
+				assert.False(t, result.Closed)
+				assert.True(t, result.Authenticated)
+				assert.True(t, result.Recorded)
+				assert.Equal(t, "shell", result.Type)
+				assert.Equal(t, "xterm", result.Term)
+				assert.InDelta(t, 1.23, result.Position.Longitude, 0.001)
+				assert.Equal(t, []string{"pty-output", "window-change"}, result.Events.Types)
+				assert.Equal(t, []int{0, 1}, result.Events.Seats)
+				assert.NotNil(t, result.Device)
+				assert.Equal(t, "ns-id-1", result.TenantID)
```
[the remark this answers is not in the document]
Comment (otavio) [reply api/store/pg/entity/session_test.go:138]: Fixed: Added `assert.InDelta(t, 4.56, result.Position.Latitude, 0.001)` assertion.

File: api/store/pg/entity/session_test.go [hunk 32]
```diff
@@ -0,0 +1,455 @@
...
+	"github.com/shellhub-io/shellhub/pkg/models"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestSessionFromModel(t *testing.T) {
+	now := time.Now()
+
+	clockMock := clockmock.NewClock(t)
+	clock.DefaultBackend = clockMock
```
[the remark this answers is not in the document]
Comment (otavio) [reply api/store/pg/entity/session_test.go:19]: Fixed: Added `t.Cleanup` to restore `clock.DefaultBackend` in both `TestSessionFromModel` and `TestActiveSessionFromModel`.

File: api/store/pg/entity/session.go [hunk 33]
```diff
@@ -59,8 +60,8 @@ func SessionFromModel(model *models.Session) *Session {
 		Term:          model.Term,
 		Longitude:     model.Position.Longitude,
 		Latitude:      model.Position.Latitude,
-		CreatedAt:     time.Now(),
-		UpdatedAt:     time.Now(),
+		CreatedAt:     clock.Now(),
+		UpdatedAt:     clock.Now(),
```
[the remark this answers is not in the document]
Comment (otavio) [reply api/store/pg/entity/session.go]: Fixed: `SessionFromModel` now calls `clock.Now()` once and reuses the value for both `CreatedAt` and `UpdatedAt`. Test mock updated from `.Twice()` to `.Once()`.

File: api/store/pg/entity/session_test.go (hunk 32 above)
[the remark this answers is not in the document]
Comment (otavio) [reply api/store/pg/entity/session_test.go:19]: Fixed: Added `t.Cleanup` to restore the original `clock.DefaultBackend` after each test.

File: api/store/pg/entity/session_test.go [hunk 34]
```diff
@@ -0,0 +1,455 @@
...
+			tt.check(t, result)
+		})
+	}
+}
+
+func TestActiveSessionFromModel(t *testing.T) {
+	now := time.Now()
+
+	clockMock := clockmock.NewClock(t)
+	clock.DefaultBackend = clockMock
```
[the remark this answers is not in the document]
Comment (otavio) [reply api/store/pg/entity/session_test.go:181]: Fixed: Added `t.Cleanup` to restore `clock.DefaultBackend`.

File: api/store/pg/entity/user_test.go [hunk 35]
```diff
@@ -0,0 +1,252 @@
...
+				Origin:   models.UserOriginSAML,
+				Status:   models.UserStatusNotConfirmed,
+				UserData: models.UserData{},
+				Password: models.UserPassword{},
+				MFA:      models.UserMFA{},
+				Preferences: models.UserPreferences{
+					AuthMethods: []models.UserAuthMethod{},
+				},
+			},
+		},
```
[the remark this answers is not in the document]
Comment (otavio) [reply api/store/pg/entity/user_test.go]: Fixed: Renamed to "SAML origin with non-confirmed status and empty auth methods".

File: api/store/pg/entity/device_test.go [hunk 36]
```diff
@@ -0,0 +1,271 @@
...
+				assert.InDelta(t, 4.56, result.Latitude, 0.001)
+				assert.Equal(t, "device-info-id", result.Identifier)
+				assert.Equal(t, "My Device", result.PrettyName)
+				assert.Equal(t, "1.0.0", result.Version)
+				assert.Equal(t, "amd64", result.Arch)
+				assert.Equal(t, "linux", result.Platform)
+				assert.Equal(t, now, result.CreatedAt)
+				assert.Equal(t, now, result.LastSeen)
+				assert.True(t, result.UpdatedAt.IsZero())
+			},
```
[the remark this answers is not in the document]
Comment (otavio) [reply api/store/pg/entity/device_test.go:60]: Acknowledged. `RemovedAt` is intentionally not mapped by `DeviceFromModel` — it's managed at the database level (soft delete). The field being nil-by-default documents the behavior implicitly.

File: api/store/pg/entity/device_test.go [hunk 37]
```diff
@@ -0,0 +1,271 @@
...
+			name: "nil Tags",
+			entity: &Device{
+				ID:     "device-uid-6",
+				Status: "accepted",
+				Tags:   nil,
+			},
+			check: func(t *testing.T, result *models.Device) {
+				assert.Empty(t, result.Tags)
+				assert.Nil(t, result.TagIDs)
+			},
```
[the remark this answers is not in the document]
Comment (otavio) [reply api/store/pg/entity/device_test.go:268]: Keeping both test cases intentionally. While they exercise the same code path, they document two distinct input scenarios (empty slice vs nil) explicitly, which is useful for anyone reading the tests.

File: api/store/pg/entity/device_test.go [hunk 38]
```diff
@@ -0,0 +1,273 @@
...
+				Status: models.DeviceStatusAccepted,
+				Taggable: models.Taggable{
+					Tags: []models.Tag{
+						{ID: "tag-1", Name: "prod", TenantID: "t1"},
+						{ID: "tag-2", Name: "staging", TenantID: "t1"},
+					},
+				},
+			},
+			check: func(t *testing.T, result *Device) {
+				assert.Len(t, result.Tags, 2)
```
[the remark this answers is not in the document]
Comment (otavio) [reply api/store/pg/entity/device_test.go]: Fixed: changed `assert.Len` to `require.Len` before all index-based access in `device_test.go` (lines 103, 119, 201, 203). Also added missing `Info` sub-field assertions (PrettyName, Version, Arch, Platform) in the DeviceToModel 'full fields' test.

File: api/store/pg/entity/public-key_test.go [hunk 39]
```diff
@@ -0,0 +1,171 @@
...
+				},
+			},
+			check: func(t *testing.T, result *PublicKey) {
+				assert.Equal(t, "tenant-id-1", result.NamespaceID)
+				assert.Equal(t, "SHA256:abc123", result.Fingerprint)
+				assert.Equal(t, []byte("ssh-rsa AAAA..."), result.Data)
+				assert.Equal(t, "my-key", result.Name)
+				assert.Equal(t, "*.example.com", result.FilterHostname)
+				assert.Equal(t, now, result.CreatedAt)
+				assert.Len(t, result.Tags, 2)
```
[the remark this answers is not in the document]
Comment (otavio) [reply api/store/pg/entity/public-key_test.go]: Fixed: changed `assert.Len` to `require.Len` before all index-based access in `public-key_test.go` (lines 46, 67, 141, 143).

File: api/store/pg/entity/device_test.go [hunk 40]
```diff
@@ -0,0 +1,278 @@
...
+				assert.InDelta(t, 1.23, result.Position.Longitude, 0.001)
+				assert.InDelta(t, 4.56, result.Position.Latitude, 0.001)
+				assert.NotNil(t, result.Info)
+				assert.Equal(t, "info-id", result.Info.ID)
+				assert.Equal(t, "My Device", result.Info.PrettyName)
+				assert.Equal(t, "1.0.0", result.Info.Version)
+				assert.Equal(t, "amd64", result.Info.Arch)
+				assert.Equal(t, "linux", result.Info.Platform)
+				assert.NotNil(t, result.Identity)
+				assert.Equal(t, "00:11:22:33:44:55", result.Identity.MAC)
```
[the remark this answers is not in the document]
Comment (otavio) [reply api/store/pg/entity/device_test.go]: Fixed: changed `assert.NotNil` to `require.NotNil` before all pointer dereferences in `device_test.go`, `system_test.go`, and `namespace_test.go`.
